### PR TITLE
Existence check root to db

### PR DESCRIPTION
**Pull request by SaravShah**
- includes ```check_online_to_db_existence``` class method, unfinished RSpec tests, and in progress benchmarking 

TO DO:
- clean up/ finish rspec tests ✔️ 
- clean up commit history ✔️ 
- clean up benchmarking code ✔️ 

Note:
- In RSpec test line 18, I used ```any_args``` to represent the ```storage_root_current_size ``` because TravisCI and my local machine had differing values. Not sure if we should figure out a way to make both Travis and local RSpec tests happy, or leave it as ```any_args```.

- Also RuboCop gave me an error for using ``` to receive(:update_or_create)```. 
RuboCop error message:
```spec/lib/audit/storage_root_to_db_spec.rb:29:31: C: RSpec/MessageSpies: Prefer have_received for setting message expectations. Setup po_handler as a spy using allow or instance_spy.```

-- Here is the original test below: 
```  it "should call 'update_or_create' with the expected values" do 
    expected_argument_list = [
      { druid: 'bj102hs9687', storage_root_current_version: 3 },
      { druid: 'bp628nk4868', storage_root_current_version: 1 },
      { druid: 'bz514sm9647', storage_root_current_version: 3 },
      { druid: 'dc048cw1328', storage_root_current_version: 2 },
      { druid: 'jj925bx9565', storage_root_current_version: 2 }
    ]
    expected_argument_list.each do |arg_hash|
      po_handler = instance_double(PreservedObjectHandler)
      allow(PreservedObjectHandler).to receive(:new).with(arg_hash[:druid], arg_hash[:storage_root_current_version], any_args).and_return(po_handler)
      expect(po_handler).to receive(:update_or_create)
    end
    subject    
  end
 ```

-- This PR has the updated test which satisfies rubocop but it is pretty ugly, let me know if you guys like the test or if I should use my old test ^^^ and silence rubocop!

- Also had to add a ``` PreservationPolicy.seed_from_config``` to RSpec tests to get the ```:update_or_create``` method working, so another reminder to figure out if we should seed the db before the tests are run!

PS: I will continue to work on the benchmarking - I am currently working on that file still. 👍 

connects #107

### CODE REVIEW

File: config/settings.yml [hunk 1]
```diff
@@ -1,3 +1,5 @@
 moab:
-  storage_roots: 'spec/fixtures'
-  storage_trunk: 'moab_storage_root'
+  # storage_roots: 'spec/fixtures'
+  # storage_trunk: 'moab_storage_root'
+  storage_roots: ['/services-disk11']
+  storage_trunk: 'sdr2objects'
```
Comment (ndushay) [inline config/settings.yml]: when the time comes to take off WIP, your local settings should NOT be included.  The actual way to do this is with a file `config/settings.local.yml` file -- the settings in that file take precedence over these, and we have this in .gitignore:

```
config/settings.local.yml
config/settings/*.local.yml
config/environments/*.local.yml
```

File: lib/validations/storage_root_to_db.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+class StorageRootToDB
```
Comment (ndushay) [inline lib/validations/storage_root_to_db.rb]: I know this is WIP;  just noting that we thought `lib/audit` was a better directory name at our meeting this afternoon 2017-10-3

File: spec/fixtures/empty/.keep [hunk 3]
```diff
@@ -0,0 +1 @@
+.keep
```
[the remark this answers is not in the document]
Comment (SaravShah) [reply spec/fixtures/empty/.keep]: This folder is required for one of the RSpec tests to pass.  The case in which the 'storage directory exists but it is empty'.

File: lib/audit/storage_root_to_db.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+##
+# Method that will check the online service disks for the existence of moabs in postgres database
```
[the remark this answers is not in the document]
Comment (SaravShah) [reply lib/audit/storage_root_to_db.rb]: I actually didn't even think of that. I set the method up to check a single storage_dir. But that does bring up the question, should this method be able to take in a list of storage_dirs, or should we have whatever is calling this method take responsibility for that?

**Comment by ndushay** [comment]
What if we have another ticket for creating a second method that loops through all the storage roots in the Settings file?  (and runs for-ev-ah)

**Comment by SaravShah** [comment]
I like that idea!

**Pull request merged**